src/micro_helmet/provider.es

```diff
@@ -0,0 +1,48 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import { Provider } from './context';
+import { parseProps, collectProps } from './utils';
+
+
+class MicroHelmetProvider extends PureComponent {
+  constructor(props) {
+    super(props);
+    this.propsArray = [];
+  }
+
+  generateContext() {
+    const { context } = this.props;
+    context.value = parseProps(this.propsArray.reduce(collectProps, {}));
+    if (process.browser) this.onChangeInBrowser();
```

Comment:
let's change this to something browser-agnostic, like 

```js
// Below imports, a one-off
const isServerEnv = (
  typeof process !== 'undefined' && process && process.versions && process.versions.node
);
```

```js
// here
if (!isServerEnv) this.updateBrowserTitle();
```

also since `this.updateBrowserTitle` will be called 2-3 times repeatedly, debounce it. We can't debounce on the server, but the client is fine.